src/GurobiSolverInterface.jl

```diff
@@ -334,6 +310,8 @@ const rev_var_type_map = Compat.@compat Dict(
 )
 
 function setvartype!(m::GurobiMathProgModel, vartype::Vector{Symbol})
+    # do this to make sure we deal with new columns
+    updatemodel!(m)
```

Comment:
Irrespective of this PR, I think this is a bugfix that our current JuMP-level behavior is masking.